ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/GasLimitCalculator.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.blockcreation;
+
+import java.util.function.LongUnaryOperator;
+
+public interface GasLimitCalculator extends LongUnaryOperator {
```

Comment:
Not a fan of the name that LongUnaryOperator forces sub-classes to use (`applyAsLong`) - it's not very readable and clear as to what is being overridden.  Do we really need the syntactic sugar features that the functional comosition provides?  i.e. will we ever actually use them?

For readability I would prefer a non-extending interface with one method `calculateNextGasTarget` so it reads clearly what is going on in subclasses.

Reply:
Non-extending interface sounds good to me 👍